### ISSUE TITLE
When an extension is adding multiple times on a page, only one gets rendered in Visual Builder.  

### ISSUE THREAD

**Issue by MaryJoStaebler** [elegantthemes/create-divi-extension#319]
### Problem Description
When an extension is adding multiple times on a page, only one gets rendered in Visual Builder.  

I have a few modules in my plugin.  Only in Visual Builder mode, does the "Oops! An Error has occurred" for one of a few instances of the same module.  The computed property for the react module comes back as null, but only for all but one instance of that module.  Sometimes it's if multiple modules from this one plugin is on the page.  

The non-VB version works fine.

**Comment by xxtesaxx** [comment]
You most likely have some errors in the way your jsx file is implemented. It would help a great deal if you could share your code. Also the code for your computed propery would be helpful.

**Comment by MaryJoStaebler** [comment]
Here's the JSX
```
// External Dependencies
import React, { Component } from 'react';

// Internal Dependencies
import './style.css';


class SinglePropertyDetails extends Component {

  static slug = 'cv_single_property_details';

  constructor(){
    super();
    this.state = {a: false}
  }
  componentDidUpdate(prevProps) {
    console.log('update',this);
    // Initialization/re-initialization should be done only if computed fields has been changed
    if (prevProps.__details !== this.props.__details) {
      console.log('here',this);

    }
  }
  render() {
    console.log('there',this);
    return <ul className={"property-details"}>
      {/* <li>{this.props.__details.bedroom} BED</li>
      <li>{this.props.__details.bathroom} BATH</li>
      <li>{this.props.__details.square_feet} SQ FT</li> */}
   </ul>;
  }

}

export default SinglePropertyDetails;



```



Here's the PHP
```
<?php

class CV_SinglePropertyDetails extends ET_Builder_Module {

	public $slug       = 'cv_single_property_details';
	public $vb_support = 'on';

	protected $module_credits = array(
		'module_uri' => '',
		'author'     => 'Mary Jo Staebler',
		'author_uri' => 'maryjostaebler.com',
	);

	public function init() {
		$this->name = esc_html__( 'Villas Property Details', 'cv-custom-harlow-builders' );
	}

	public function get_fields() {
		return array(
			''
			'__details' => array(
				'type' => 'computed',
				'computed_callback' => array( 'CV_SinglePropertyDetails', 'get_property_details' ),
			),
		);
	}

	static function get_property_details( $args = array(), $conditional_tags = array(), $current_page = array() ) {

		$post = get_post( $this->get_the_ID() );

		$property_details =array (
			'bedroom' => get_post_meta($post->ID, "property_bedroom", true),
			'bathroom' => get_post_meta($post->ID, "property_bathroom", true),
			'square_feet' => get_post_meta($post->ID, "property_square_feet", true),
		);
		return 'blah';
	}



	public function render( $attrs, $content = null, $render_slug ) {
		global $post;


		$propertybedroom 	= get_post_meta($post->ID, "property_bedroom", true);
		$propertybathroom 	= get_post_meta($post->ID, "property_bathroom", true);
		$propertysquarefeet = get_post_meta($post->ID, "property_square_feet", true);

		ob_start();

		?>
			<ul class="property-details">
				<?php if (strlen($propertybedroom) > 0) { 
					?><li><?echo esc_html($propertybedroom);?> BED</li><?php
				}?>
					<?php if (strlen($propertybathroom) > 0) { 
					?><li><?echo esc_html($propertybathroom);?> BATH</li><?php
				}?>
					<?php if (strlen($propertysquarefeet) > 0) { 
					?><li><?echo esc_html($propertysquarefeet);?> SQ FT </li><?php
				}?>
			</ul>
		<?php
		$property_details = ob_get_contents();

		ob_end_clean();


		return $property_details;
	}
}

new CV_SinglePropertyDetails;
```

**Comment by xxtesaxx** [comment]
You are calling $this inside the static function for the computed property. You should never call $this inside a static function.

**Comment by MaryJoStaebler** [comment]
Ended up that I needed to have a 'computed_depends_on' to make it work.  So I added a text field for a field even though I don't need it.

```
public function get_fields() {
		return array(
			'nothing' => array(
				'label'             => esc_html__( 'Nothing', 'et_builder' ),
				'type'              => 'text',
				'option_category'   => 'advanced',
				'description'       => esc_html__( 'Nothing', 'et_builder' ),
				'computed_affects'   => array(
					'__details',
				),
				'default'           => '',
			),
			'__details' => array(
				'type' => 'computed',
				'computed_callback' => array( 'SinglePropertyDetails', 'get_property_details' ),
				'computed_depends_on' => array(
					'nothing',
				),
			),
		);
	}
```

 I did remove the $this. Thanks!  

Now just have to figure out how to hide this other field that I don't need to associate with the 'computed_depends_on'.   Looking for how to bury or hide.

**Comment by MaryJoStaebler** [comment]
FYI - on the updated get_property_details function
```
static function get_property_details( $args = array(), $conditional_tags = array(), $current_page = array() ) {


		$defaults = array(
			'nothing'                     => '',
		);

		$args = wp_parse_args( $args, $defaults );
		$post = get_post( $current_page['id']);

		$property_details =array (
			'bedroom' => get_post_meta($post->ID, "property_bedroom", true),
			'bathroom' => get_post_meta($post->ID, "property_bathroom", true),
			'square_feet' => get_post_meta($post->ID, "property_square_feet", true),
		);
		return $property_details;
	}
```

**Comment by xxtesaxx** [comment]
There is a type called "hidden" i believe. Maybe it works with this instead of adding a useless textfield.

**Comment by MaryJoStaebler** [comment]
I found how to place it under the Advanced Tab and suddenly it disappeared.  I tried the hidden and same result happened.

The initial error that I had written about was because over the past year a change in the interface occurred that the data in the computed call was not initially available for the react component (i.e was null).  Only until a few requests later did the data actually appear in a call.  So I added a check to put a placeholder of data until the data was ready.

TL;DR - data wasn't available in the React component until a few requests later.  Once the component had an error, no other requests for data was taken. 

This is why some times one instance of the module actually worked and others did not. It was a race condition.
